pkg/tls/tls.go

```diff
@@ -276,6 +299,12 @@ func (p *Provider) fetchCertificate(ctx context.Context) (time.Time, error) {
 	// If we are not using a custom root CA, then overwrite the existing with
 	// what was responded.
 	if len(p.opts.RootCACertFile) == 0 {
+		// If the root CAs bundle has been changed, send event to subscribers.
+		if !bytes.Equal(p.rootCA, bundle.CA) {
```

Comment:
Question: do we not want to send these event _after_ lines 315-322 where we could potentially error out of this func if the root CA was not decodeable?